From: Prarit Bhargava <prarit@redhat.com>
Date: Mon, 10 May 2010 18:52:35 -0400
Subject: [scsi] compat header cleanups
Message-id: <20100510185019.2781.28794.sendpatchset@prarit.bos.redhat.com>
Patchwork-id: 24982
O-Subject: [RHEL5 PATCH 22/27] compat.h cleanup: scsi changes
Bugzilla: 546740
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

>From 43a14fec7875151fca485625badb967a1e99924b Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Mon, 10 May 2010 09:50:27 -0400
Subject: [PATCH] scsi changes

The changes in scsi are NOT due to the elimination of compat files in
the scsi driver.  They are a result of modifications made elsewhere
in the kernel.

Resolves 546740.

diff --git a/drivers/scsi/be2iscsi/be.h b/drivers/scsi/be2iscsi/be.h
index 00df7d7..4200764 100644
--- a/drivers/scsi/be2iscsi/be.h
+++ b/drivers/scsi/be2iscsi/be.h
@@ -23,7 +23,6 @@
 #include <scsi/iscsi_compat2.h>
 #include <scsi/iscsi_proto2.h>
 #include <linux/blk-iopoll.h>
-#include "be_compat.h"
 #define FW_VER_LEN	32
 #define MCC_Q_LEN	128
 #define MCC_CQ_LEN	256
diff --git a/drivers/scsi/be2iscsi/be_compat.h b/drivers/scsi/be2iscsi/be_compat.h
index 1a78710..a92e41e 100644
--- a/drivers/scsi/be2iscsi/be_compat.h
+++ b/drivers/scsi/be2iscsi/be_compat.h
@@ -18,9 +18,6 @@
 #ifndef BE_COMPAT_H
 #define BE_COMPAT_H
 
-#define PTR_ALIGN(p, a)         	((typeof(p))			\
-					ALIGN((unsigned long)(p), (a)))
-
 #define DEFINE_PCI_DEVICE_TABLE(_table) struct pci_device_id _table[] 	\
 						__devinitdata
 
diff --git a/drivers/scsi/be2iscsi/be_main.c b/drivers/scsi/be2iscsi/be_main.c
index 8772f6e..445e318 100644
--- a/drivers/scsi/be2iscsi/be_main.c
+++ b/drivers/scsi/be2iscsi/be_main.c
@@ -416,7 +416,7 @@ static void hwi_ring_eq_db(struct beiscsi_hba *phba,
  * @irq: Not used
  * @dev_id: Pointer to host adapter structure
  */
-static irqreturn_t be_isr_mcc(int irq, void *dev_id)
+static irqreturn_t be_isr_mcc(int irq, void *dev_id, struct pt_regs *pt_regs)
 {
 	struct beiscsi_hba *phba;
 	struct be_eq_entry *eqe = NULL;
@@ -463,7 +463,7 @@ static irqreturn_t be_isr_mcc(int irq, void *dev_id)
  * @irq: Not used
  * @dev_id: Pointer to host adapter structure
  */
-static irqreturn_t be_isr_msix(int irq, void *dev_id)
+static irqreturn_t be_isr_msix(int irq, void *dev_id, struct pt_regs *pt_regs)
 {
 	struct beiscsi_hba *phba;
 	struct be_eq_entry *eqe = NULL;
@@ -523,7 +523,7 @@ static irqreturn_t be_isr_msix(int irq, void *dev_id)
  * @irq: Not used
  * @dev_id: Pointer to host adapter structure
  */
-static irqreturn_t be_isr(int irq, void *dev_id)
+static irqreturn_t be_isr(int irq, void *dev_id, struct pt_regs *pt_regs)
 {
 	struct beiscsi_hba *phba;
 	struct hwi_controller *phwi_ctrlr;
diff --git a/drivers/scsi/fcoe/fcoe.c b/drivers/scsi/fcoe/fcoe.c
index d789c32..b7d2e91 100644
--- a/drivers/scsi/fcoe/fcoe.c
+++ b/drivers/scsi/fcoe/fcoe.c
@@ -37,6 +37,7 @@
 #include <linux/rtnetlink.h>
 
 #include <scsi/fc/fc_encaps.h>
+#include <scsi/fc/fc_fcoe.h>
 #include <scsi/fc/fc_fip.h>
 
 #include <scsi/libfc.h>
diff --git a/drivers/scsi/fnic/vnic_dev.h b/drivers/scsi/fnic/vnic_dev.h
index f9935a8..e857ebb 100644
--- a/drivers/scsi/fnic/vnic_dev.h
+++ b/drivers/scsi/fnic/vnic_dev.h
@@ -66,19 +66,6 @@
 #define VNIC_PADDR_TARGET	0x0000000000000000ULL
 #endif
 
-#ifndef readq
-static inline u64 readq(void __iomem *reg)
-{
-	return ((u64)readl(reg + 0x4UL) << 32) | (u64)readl(reg);
-}
-
-static inline void writeq(u64 val, void __iomem *reg)
-{
-	writel(val & 0xffffffff, reg);
-	writel(val >> 32, reg + 0x4UL);
-}
-#endif
-
 enum vnic_dev_intr_mode {
 	VNIC_DEV_INTR_MODE_UNKNOWN,
 	VNIC_DEV_INTR_MODE_INTX,
diff --git a/drivers/scsi/libfc/fc_disc.c b/drivers/scsi/libfc/fc_disc.c
index 9b0a519..912592c 100644
--- a/drivers/scsi/libfc/fc_disc.c
+++ b/drivers/scsi/libfc/fc_disc.c
@@ -34,6 +34,7 @@
 
 #include <linux/timer.h>
 #include <linux/err.h>
+#include <linux/workqueue.h>
 #include <asm/unaligned.h>
 
 #include <scsi/fc/fc_gs.h>
diff --git a/drivers/scsi/libfc/fc_exch.c b/drivers/scsi/libfc/fc_exch.c
index 2dad7f3..844922f 100644
--- a/drivers/scsi/libfc/fc_exch.c
+++ b/drivers/scsi/libfc/fc_exch.c
@@ -27,6 +27,7 @@
 #include <linux/gfp.h>
 #include <linux/err.h>
 #include <linux/mempool.h>
+#include <linux/workqueue.h>
 
 #include <scsi/fc/fc_fc2.h>
 
@@ -358,7 +359,7 @@ static int fc_exch_done_locked(struct fc_exch *ep)
 
 	if (!(ep->esb_stat & ESB_ST_REC_QUAL)) {
 		ep->state |= FC_EX_DONE;
-		if (cancel_delayed_work(&ep->timeout_work))
+		if (cancel_delayed_work(&ep->timeout_work.work))
 			atomic_dec(&ep->ex_refcnt); /* drop hold for timer */
 		rc = 0;
 	}
@@ -1621,7 +1622,7 @@ static void fc_exch_reset(struct fc_exch *ep)
 
 	spin_lock_bh(&ep->ex_lock);
 	ep->state |= FC_EX_RST_CLEANUP;
-	if (cancel_delayed_work(&ep->timeout_work))
+	if (cancel_delayed_work(&ep->timeout_work.work))
 		atomic_dec(&ep->ex_refcnt);	/* drop hold for timer */
 	resp = ep->resp;
 	ep->resp = NULL;
@@ -2013,7 +2014,7 @@ static void fc_exch_els_rrq(struct fc_seq *sp, struct fc_frame *fp)
 		atomic_dec(&ep->ex_refcnt);	/* drop hold for rec qual */
 	}
 	if (ep->esb_stat & ESB_ST_COMPLETE) {
-		if (cancel_delayed_work(&ep->timeout_work))
+		if (cancel_delayed_work(&ep->timeout_work.work))
 			atomic_dec(&ep->ex_refcnt);	/* drop timer hold */
 	}
 
diff --git a/include/scsi/fc_compat.h b/include/scsi/fc_compat.h
index 4496623..97dffc8 100644
--- a/include/scsi/fc_compat.h
+++ b/include/scsi/fc_compat.h
@@ -30,31 +30,6 @@ static inline struct page *sg_page(struct scatterlist *sg)
 
 #define BIT(nr) (1UL << (nr))
 
-#define MAC_FMT "%02x:%02x:%02x:%02x:%02x:%02x"
-#define MAC_BUF_SIZE 18
-#define DECLARE_MAC_BUF(var) char var[MAC_BUF_SIZE]
-
-static size_t _format_mac_addr(char *buf, int buflen,
-                                const unsigned char *addr, int len)
-{
-	int i;
-	char *cp = buf;
-
-	for (i = 0; i < len; i++) {
-		cp += scnprintf(cp, buflen - (cp - buf), "%02x", addr[i]);
-		if (i == len - 1)
-			break;
-		cp += strlcpy(cp, ":", buflen - (cp - buf));
-	}
-	return cp - buf;
-}
-
-static inline char *print_mac(char *buf, const unsigned char *addr)
-{
-	_format_mac_addr(buf, MAC_BUF_SIZE, addr, ETH_ALEN);
-	return buf;
-}
-
 #define dev_get_by_name(_inet, _name)  dev_get_by_name(_name)
 
 #define vlan_dev_real_dev(_ndev) VLAN_DEV_INFO(_ndev)->real_dev
@@ -73,22 +48,6 @@ static inline char *print_mac(char *buf, const unsigned char *addr)
 
 #define flush_work(_wk) flush_scheduled_work()
 
-struct delayed_work {
-        struct work_struct work;
-};
-
-static inline int cancel_delayed_work_sync(struct delayed_work *dwork)
-{
-	int ret;
-
-	ret = cancel_delayed_work(&dwork->work);
-	if (!ret)
-		flush_scheduled_work();
-	return ret;
-}
-
-#define cancel_delayed_work(_dwork) cancel_delayed_work(&(_dwork)->work)
-
 static inline int schedule_delayed_work_compat(struct delayed_work *work,
 					       unsigned long delay)
 {
@@ -112,6 +71,7 @@ static inline int queue_delayed_work_compat(struct workqueue_struct *wq,
 
 #undef INIT_WORK
 #define INIT_WORK(_work, _func) INIT_WORK_compat(_work, _func)
+#undef INIT_DELAYED_WORK
 #define INIT_DELAYED_WORK(_work,_func) INIT_WORK(&(_work)->work, _func)
 
 #define queue_delayed_work queue_delayed_work_compat
diff --git a/include/scsi/iscsi_compat2.h b/include/scsi/iscsi_compat2.h
index 726606e..f68c6fb 100644
--- a/include/scsi/iscsi_compat2.h
+++ b/include/scsi/iscsi_compat2.h
@@ -121,6 +121,7 @@ static inline void INIT_WORK_compat(struct work_struct *work, void *func)
 
 #undef INIT_WORK
 #define INIT_WORK(_work, _func) INIT_WORK_compat(_work, _func)
+#undef INIT_DELAYED_WORK
 #define INIT_DELAYED_WORK(_work,_func) INIT_WORK_compat(_work, _func)
 
 #endif
